### PR TITLE
feat: added acm: config sync and policy controller

### PR DESCRIPTION
**Pull request by yliaog**

### CODE REVIEW

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  fleet_membership_re = "//gkehub.googleapis.com/(.*)$"
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:18]: nit: not used

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  depends_on = [
+    google_project_service.acm
+  ]
+}
+
+resource "google_gke_hub_feature_membership" "acm_feature_member" {
+  for_each = toset(var.cluster_membership_ids)
+
+  project    = var.project_id
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:76]: Most of these `project_id` should probably now be `fleet_project_id`

```suggestion
  project    = var.fleet_project_id
```

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  location   = "global"
+  feature    = "configmanagement"
+  membership = regex(local.membership_re, each.key)[2]
+  configmanagement {
+    version = "1.15.0"
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:81]: nit: do we want the current release?

```suggestion
    version = "1.17.2"
```

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * limitations under the License.
+ */
+
+locals {
+  fleet_membership_re = "//gkehub.googleapis.com/(.*)$"
+}
+
+resource "google_project_service" "acm" {
+  project = var.project_id
+  service = "anthosconfigmanagement.googleapis.com"
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:23]: Would you prefer to enable when the project is created?

https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/blob/main/2-multitenant/modules/env_baseline/main.tf#L60

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+resource "google_project_service" "csr" {
+  project = var.project_id
+  service = "sourcerepo.googleapis.com"
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:28]: Would you prefer to enable when the project is created?

https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/blob/main/2-multitenant/modules/env_baseline/main.tf#L60

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  depends_on = [
+    google_project_service.acm
+  ]
+}
+
+resource "google_gke_hub_feature_membership" "acm_feature_member" {
+  for_each = toset(var.cluster_membership_ids)
+
+  project    = var.project_id
+  location   = "global"
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/acm.tf:77]: FYI: The cluster memberships are regional and can be retrieved from their `cluster_membership_ids`.

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 1 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:18]: removed.

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 2 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:76]: done

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 4 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:23]: yes, moved to activation at project creation time

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+resource "google_service_account" "root-reconciler" {
+  project      = var.project_id
+  account_id   = "root-reconciler"
+  display_name = "root-reconciler"
+}
+
+resource "google_project_iam_member" "root-reconciler" {
+  for_each = toset(["roles/source.reader"])
```
[the remark this answers is not in the document]
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:47]: done

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 6 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:77]: done

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 3 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf:62]: done

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 5 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf]: done

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 6 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf]: done

**Comment by apeabody** [comment]
Triggering the test:

```
Error: Error deleting FeatureMembership: googleapi: Error 409: unable to queue the operation
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.ResourceInfo",
    "resourceName": "projects/eab-fleet-non-production-2aca/locations/global/features/configmanagement"
  }
]
}
```

**Comment by apeabody** [comment]
> Triggering the test:
> 
> ```
> Error: Error deleting FeatureMembership: googleapi: Error 409: unable to queue the operation
> Details:
> [
>   {
>     "@type": "type.googleapis.com/google.rpc.ResourceInfo",
>     "resourceName": "projects/eab-fleet-non-production-2aca/locations/global/features/configmanagement"
>   }
> ]
> }
> ```

Looks like it was a flake

File: 4-fleetscope/modules/env_baseline/README.md [hunk 8]
```diff
@@ -13,7 +13,7 @@ The following resources are created:
 |------|-------------|------|---------|:--------:|
 | cluster\_membership\_ids | The membership IDs in the scope | `list(string)` | n/a | yes |
 | env | The environment to prepare (ex. development) | `string` | n/a | yes |
-| fleet\_project\_id | The fleet project ID | `string` | n/a | yes |
+| fleet\_project\_id | The Fleet project ID | `string` | n/a | yes |
```
Comment (apeabody) [inline 4-fleetscope/modules/env_baseline/README.md:16]: Given the prevalence of `fleet`, we should probably be consistent with caps or no-caps.  From the docs usage I would vote for lowercase: https://cloud.google.com/kubernetes-engine/docs/fleets-overview

Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/README.md]: yes, agreed, switched to lowercase.

**Comment by apeabody** [comment]
Look like a flakey build, re-triggering

**Comment by apeabody** [comment]
`Error: Error deleting FeatureMembership: googleapi: Error 409: unable to queue the operation`

**Comment by apeabody** [comment]
> `Error: Error deleting FeatureMembership: googleapi: Error 409: unable to queue the operation`

Let's see if we can retry these automatically: https://github.com/GoogleCloudPlatform/terraform-google-enterprise-application/pull/44

File: 4-fleetscope/modules/env_baseline/acm.tf [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_sourcerepo_repository" "acm_repo" {
+  project = var.fleet_project_id
+  name    = "eab-acm"
+}
+
+resource "google_service_account" "root-reconciler" {
```
Comment (gtsorbo) [inline 4-fleetscope/modules/env_baseline/acm.tf:22]: nit: here and throughout, can we standardize on underscores for resource and module names

**Review by gtsorbo — APPROVED** [review]
minor feedback but otherwise LGTM, thanks!

File: 4-fleetscope/modules/env_baseline/acm.tf (hunk 9 above)
Comment (yliaog) [reply 4-fleetscope/modules/env_baseline/acm.tf]: done

**Pull request merged**